### ISSUE TITLE
Update Readme.md File

### ISSUE THREAD

**Issue by yogeshAgarwal** [TheAlgorithms/Java#304]

**Comment by varunu28** [comment]
@yogeshAgarwal Can you please some details on why have you have opened this issue?

**Comment by yogeshAgarwal** [comment]
there are so many algos in this repo but only some of them is mention in the readme.md, so we should design redme.md so that it can gives all the information of algos present in the repo.

**Comment by varunu28** [comment]
If you want to take up on this then you can create a new markdown file with a list of all algos with their links under their correct category. Let me know if you want to take this

**Comment by yogeshAgarwal** [comment]
yes, i want to do work on it but after 29 November, is it okay?

**Comment by varunu28** [comment]
Yes sure. I will assign this to you

**Comment by GammaBurst101** [comment]
@varunu28 , Did @yogeshAgarwal complete the work? If not, then can I do it?

**Comment by varunu28** [comment]
@GammaBurst101 Yes you can take it up

**Comment by GammaBurst101** [comment]
@varunu28 , How should I do it? Write the name of each .java file under their respective headings like "Dynamic Programming"and a relative link in markdown after the already written explanations of the Sorting and Searching algorithms? Or should I do it in any other way?

**Comment by varunu28** [comment]
@yogeshAgarwal Yes. I think creating a tabular format for representing all the algorithms would be much more representable
@dynamitechetan What is your opinion about this?

**Comment by dynamitechetan** [comment]
Yes. Tabular format will be good.

**Comment by varunu28** [comment]
@yogeshAgarwal You can go ahead with the changes and make a PR

**Comment by GammaBurst101** [comment]
@varunu28 Now, I feel that I can't do it. You were talking about tabular format and I don't know how to do that. I thought of a list but that also I can't do as the markdown written by me doesn't work. (Example: [bags](Data Structures/Bags/Bag.java)). 
I joined GitHub just 12 days ago. I think I should first learn markdown and other things before venturing out like this.
Sorry for troubling you. Please do it either yourself or assign someone else.

**Comment by varunu28** [comment]
@GammaBurst101 It is alright. Getting hold of README syntax won't take more than a day for you. Just go through this [README Guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet).

See what I did there :+1:

**Comment by GammaBurst101** [comment]
@varunu28 thanks. I made a PR. Please take a look. #377
